### PR TITLE
Remove officially unmaintained collections

### PR DESCRIPTION
**Pull request by mariolenz**
##### SUMMARY
The current process to identify and remove an unmaintained collection looks very much focused on _abandoned_ collections. But I think we should have a way to remove a collection that announces itself as being unmaintained quicker.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
removal_from_ansible.rst

##### ADDITIONAL INFORMATION
ansible-community/community-topics#124

cc @felixfontein

### CODE REVIEW

**Comment by felixfontein** [comment]
I marked this as WIP since this needs to be voted on before it can be merged.

**Comment by mariolenz** [comment]
> I marked this as WIP since this needs to be voted on before it can be merged.

I've added ansible-community/community-topics#130. Maybe we can discuss this tomorrow and, if people agree, I'll start a vote.

File: removal_from_ansible.rst
Comment (felixfontein) [inline removal_from_ansible.rst:133]: The goal of referring to the general processes inside this document was to avoid repeating these instructions everywhere.

Comment (mariolenz) [reply removal_from_ansible.rst:133]: I understand, but referring to `announce removal` and `remove collection` looks like there are two separate steps. I think we can do it in just one step. The process I propose is not simply `announce removal` + `remove collection` ;-)

Additionally, `Announcing upcoming removal` says to `announce upcoming removal in the collection's issue tracker`. Which doesn't make sense if the collection is officially unmaintained. And, actually, wouldn't even be possible if the collection has been archived.

Comment (flowerysong) [reply removal_from_ansible.rst:133]: I don't see the point of two divergent processes for doing the exact same thing, or duplicating documentation details in order to do so. If you think that in some circumstances announcing it on the collection's issue tracker won't be possible, just add an "if possible" on the end of the existing documentation for that step.

Comment (mariolenz) [reply removal_from_ansible.rst]: @felixfontein @flowerysong What do you think about my latest commit? Does it look OK to you now?

File: removal_from_ansible.rst
Comment (Andersson007) [inline removal_from_ansible.rst:131]:
```suggestion
See `the example pull request <https://github.com/ansible-community/ansible-build-data/pull/374/files>`_ in the ``ansible-build-data`` repository to learn how to remove the collection.
```

**Comment by mariolenz** [comment]
@felixfontein @flowerysong I'm referring to the general processes again. What do you think about this PR now?

**Review by felixfontein — APPROVED** [review]
Thanks, looks good to me!

**Review by briantist — APPROVED** [review]

**Review by russoz — APPROVED** [review]
LGTM

**Review by Andersson007 — APPROVED** [review]

File: removal_from_ansible.rst
Comment (samccann) [inline removal_from_ansible.rst:129]: Picking a corner case here - What if a collection owner decides it's deprecated/unmaintained late in an Ansible development cycle?  Say foo.bar collection is deprecated a week before the Ansible 10 alpha - do we still remove it in Ansible 10?

Comment (mariolenz) [reply removal_from_ansible.rst:129]: @samccann Good point, we should clarify this.

In this case, I think we should remove the collection since it's only an alpha release. That's not late, it's early in the development cycle. But probably not once there's a pre-release that comes with a feature freeze (Ansible-10.0.0 beta1).

What do you think about:

```
If the current major release is X and there hasn't been a feature freeze release of the next major version X+1:

#. `Announce upcoming removal from the (X+1).0.0 Ansible release <announce_removal_>`_.
#. `Remove collection from the (X+1).0.0 Ansible release <remove_collection_>`_.

If there already has been a feature freeze release of the next major version X+1:

#. `Announce upcoming removal from the (X+2).0.0 Ansible release <announce_removal_>`_.
#. `Remove collection from the (X+2).0.0 Ansible release <remove_collection_>`_.
```

Comment (felixfontein) [reply removal_from_ansible.rst:129]: Or simply define a new variable `Y` which is `X+1` or `X+2`, and then use `Y`, to avoid having the list of instructions twice :)

Comment (mariolenz) [reply removal_from_ansible.rst:129]: @samccann Or, as @felixfontein suggested, simplify this using a variable `Y`:

```
If the current major release is X and there hasn't been a feature freeze release of the next major version X+1, remove the collection from Y=(X+1).0.0.
If there already has been a feature freeze release of the next major version X+1, remove the collection from Y=(X+2).0.0.

#. `Announce upcoming removal from the (Y Ansible release <announce_removal_>`_.
#. `Remove collection from the Y Ansible release <remove_collection_>`_.
```

Comment (felixfontein) [reply removal_from_ansible.rst:129]: There's now an opening bracket in the first `#.` line that shouldn't be there, but besides from that, LGTM.

**Comment by mariolenz** [comment]
@samccann What do you think about my last commit / this PR now?

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
LGTM

File: removal_from_ansible.rst
Comment (flowerysong) [inline removal_from_ansible.rst:122]:
```suggestion
Removing a collection that has been explicitly deprecated or abandoned by its former maintainers
```

File: removal_from_ansible.rst
Comment (flowerysong) [inline removal_from_ansible.rst:136]:
```suggestion
Identifying and removing an unmaintained collection that has not been deprecated by its maintainers
```

**Review by flowerysong — COMMENTED** [review]
Really just a number agreement/grammar nit, though I also tried to make the first heading's structure more like other headings at this level.

File: removal_from_ansible.rst
Comment (mariolenz) [reply removal_from_ansible.rst]: I don't like "former maintainers" here. When they deprecate or abandon the collection, they are still the maintainers. They are only _former maintainers_ afterwards.

What I mean is: Only _current_ maintainers can deprecate or abandon a collection. What do you think about my [last commit](https://github.com/ansible-collections/overview/pull/214/commits/81c1150dc3970d46f23c503e0e9822db804fd36a)?

Comment (flowerysong) [reply removal_from_ansible.rst]: It's not a big deal either way, but when this process is taking place they are the former maintainers so it feels weird to call them the maintainers. The collection currently has no maintainers, or else it would not be eligible for this process.

Comment (mariolenz) [reply removal_from_ansible.rst]: @Andersson007 @briantist @felixfontein @russoz @samccann What do you think, "former maintainers" or just "maintainers"? Personally, I don't like the "former" but I'm open to both.

Comment (felixfontein) [reply removal_from_ansible.rst]: Deprecated collections can still be maintained. Only when they are unmaintained or abandoned (is there actually a difference between the two?) they no longer have maintainers. So depending on context, both can be the right wording. So how about writing `(former) maintainers`?

**Comment by mariolenz** [comment]
The [vote on merging this](https://forum.ansible.com/t/4204/3) has been accepted.

**Pull request merged**
